### ISSUE TITLE
Another place test failures are showing up as build failures

### ISSUE THREAD

**Issue by missymessa** [dotnet/arcade#11473]
Originating PR: https://github.com/dotnet/runtime/pull/77578/checks?check_run_id=9230933216

Screenshot: 

<img width="763" alt="image" src="https://user-images.githubusercontent.com/47990216/199318858-c0cf2e58-7b36-4c3a-a709-39bf3547818e.png">

**Comment by AlitzelMendez** [comment]
I took a quick look, and this seems like a very particular scenario because we identify as a test those that report back to AzDO test results, so we remove them from the builds because we can find further information of them on AzDO test section, but on this case, there is no more information. This is the build: https://dev.azure.com/dnceng-public/public/_build/results?buildId=69285&view=results and there is not a test tab on that build so if we remove the tests from the build, we don't have more information at all, and those failures would be lost.

**Comment by missymessa** [comment]
I forgot to tag @hoyosjs on this issue. Perhaps this might be better handled by the Shared Test Infra epic instead?

**Comment by AlitzelMendez** [comment]
@missymessa yes, I think this issue doesn't belong to this epic, as the build analysis is doing the expected thing

**Comment by missymessa** [comment]
Moving to: https://github.com/dotnet/arcade/issues/6948
